providers/src/airflow/providers/amazon/aws/transfers/sftp_to_s3.py

```diff
@@ -85,6 +89,15 @@ def execute(self, context: Context) -> None:
 
         sftp_client = ssh_hook.get_conn().open_sftp()
 
+        try:
+            sftp_client.stat(self.sftp_path)
+        except FileNotFoundError:
+            if self.fail_on_file_not_exist:
+                raise
+            else:
+                self.log.info("File %s not found on SFTP server. Skipping transfer.", self.sftp_path)
+                return
```

Comment:
Just wondering about the benefit of not failing if the file does not exist.

Reply:
As mentioned in https://github.com/apache/airflow/issues/41472, the fail_on_file_not_exist option is intended to provide users greater control over the operator's behaviour, particularly in environments where continued processing is preferred.

Reply:
we have some long-time dag and need this `fail_on_file_not_exist` option to skip on file not found failure.
without this option we need to re-run long-time dag and it's really annoying 😅